rclcpp_action/src/server.cpp

```diff
@@ -461,6 +464,11 @@ ServerBase::publish_status()
 {
   rcl_ret_t ret;
 
+  // We need to hold the lock across this entire method because
+  // rcl_action_server_get_goal_handles() returns an internal pointer to the
+  // goal data.
+  std::lock_guard<std::recursive_mutex> lock(pimpl_->action_server_reentrant_mutex_);
```

Comment:
this fix is included in this PR https://github.com/ros2/rclcpp/pull/1313. https://github.com/ros2/rclcpp/blob/daa2640031598c7be32f4efb5671e544cc8b2d3e/rclcpp_action/src/server.cpp#L480

Reply:
Yeah, understood.  I was just trying to get the simpler part of that patch in, then take another look at #1313.